fv3core/grid/gnomonic.py

```diff
@@ -0,0 +1,808 @@
+import math
+
+from fv3core.utils.global_constants import PI
+
+
+def gnomonic_grid(grid_type: int, lon, lat, np):
+    """
+    Apply gnomonic grid to lon and lat arrays
+
+    args:
+        grid_type: type of grid to apply
+        lon: longitute array with dimensions [x, y]
+        lat: latitude array with dimensionos [x, y]
+    """
+    _check_shapes(lon, lat)
+    if grid_type == 0:
+        global_gnomonic_ed(lon, lat, np)
+    elif grid_type == 1:
+        gnomonic_dist(lon, lat)
+    elif grid_type == 2:
+        gnomonic_angl(lon, lat)
+    if grid_type < 3:
+        symm_ed(lon, lat)
+        lon[:] -= PI
+
+
+def _check_shapes(lon, lat):
+    if len(lon.shape) != 2:
+        raise ValueError(f"longitude must be 2D, has shape {lon.shape}")
+    elif len(lat.shape) != 2:
+        raise ValueError(f"latitude must be 2D, has shape {lat.shape}")
+    elif lon.shape[0] != lon.shape[1]:
+        raise ValueError(f"longitude must be square, has shape {lon.shape}")
+    elif lat.shape[0] != lat.shape[1]:
+        raise ValueError(f"latitude must be square, has shape {lat.shape}")
+    elif lon.shape[0] != lat.shape[0]:
+        raise ValueError(
+            "longitude and latitude must have same shape, but they are "
+            f"{lon.shape} and {lat.shape}"
+        )
+
+
+# A tile global version of gnomonic_ed
+# closer to the Fortran code
+def global_gnomonic_ed(lon, lat, np):
```

Comment:
I'm fine doing this. I  put the global gnomonic and mirror grid code into `global_setup.py`, but I did keep it in fv3core/grid because I do think it "belongs" there more than in the testing modules.